tds-itemselection-impl/src/main/java/tds/itemselection/loader/AbstractDBLoader.java

```diff
@@ -35,32 +37,53 @@
 	@Override
 	public ItemCandidatesData getItemCandidates(SQLConnection connection, UUID oppkey)
 			throws ReturnStatusException, SQLException {
-		ItemCandidatesData res = null;
-
 		SingleDataResultSet result = iSelDLL.AA_GetNextItemCandidates_SP(
-				connection, oppkey);
+				connection, oppkey, false);
 		DbResultRecord record = result.getCount() > 0 ? result.getRecords()
 				.next() : null;
+		return parseData(record, oppkey);
+	}
+
+	private ItemCandidatesData parseData(DbResultRecord record, UUID oppkey) {
+		ItemCandidatesData itemCandidatesData = null;
 		if (record != null) {
 			if (DbComparator.isEqual(
 					record.<String> get(IItemSelectionDLL.ALGORITHM),
 					IItemSelectionDLL.SATISFIED)) {
 				_logger.info("This oppkey was selected previous time");
-				res = new ItemCandidatesData(oppkey,
+				itemCandidatesData = new ItemCandidatesData(oppkey,
 						IItemSelectionDLL.SATISFIED);
 			} else {
-				res = new ItemCandidatesData(
-				record.<UUID> get(IItemSelectionDLL.OPPKEY),
-				record.<String> get(IItemSelectionDLL.ALGORITHM),
-				record.<String> get(IItemSelectionDLL.SEGMENTKEY),
-				record.<String> get(IItemSelectionDLL.SEGMENTID),
-				record.<Integer> get(IItemSelectionDLL.SEGMENT),
-				record.<String> get(IItemSelectionDLL.GROUPID),
-				record.<String> get(IItemSelectionDLL.BLOCKID),
-				record.<UUID> get(IItemSelectionDLL.SESSION),
-				record.<Boolean> get(IItemSelectionDLL.ISSIMULATION));
+				itemCandidatesData = new ItemCandidatesData(
+						record.<UUID> get(IItemSelectionDLL.OPPKEY),
+						record.<String> get(IItemSelectionDLL.ALGORITHM),
+						record.<String> get(IItemSelectionDLL.SEGMENTKEY),
+						record.<String> get(IItemSelectionDLL.SEGMENTID),
+						record.<Integer> get(IItemSelectionDLL.SEGMENT),
+						record.<String> get(IItemSelectionDLL.GROUPID),
+						record.<String> get(IItemSelectionDLL.BLOCKID),
+						record.<UUID> get(IItemSelectionDLL.SESSION),
+						record.<Boolean> get(IItemSelectionDLL.ISSIMULATION));
 			}
 		}
-		return res;
+		return itemCandidatesData;
+	}
+
+	@Override
+	public ArrayList<ItemCandidatesData> getAllItemCandidates(SQLConnection connection, UUID oppkey)
```

Comment:
If this is a new interface method we've added we should return `List` rather than ArrayList.  Callers should not know the concrete implementation of the List data structure we're using.